### PR TITLE
feat(consensus): support test for `check_gas_limit` function

### PR DESCRIPTION
**Pull request by DoTheBestToGetTheBest**

### CODE REVIEW

File: crates/consensus/common/src/validation.rs [hunk 1]
```diff
@@ -848,4 +848,78 @@ mod tests {
...
+        assert_eq!(check_gas_limit(&parent, &child, &chain_spec), Ok(()));
+    }
+
+    fn create_header(gas_limit: u64) -> SealedHeader {
+        SealedHeader { header: Header { gas_limit, ..Default::default() }, ..Default::default() }
+    }
+
+    #[test]
+    fn test_invalid_gas_limit_increase_exceeding_limit() {
+        let parent = create_header(1024 * 10);
```
Comment (Rjected) [inline crates/consensus/common/src/validation.rs:889]: not sure we need a method just for this, we can probably create the headers directly with the specified gas limits, like with the first few tests

File: crates/consensus/common/src/validation.rs [hunk 2]
```diff
@@ -848,4 +848,78 @@ mod tests {
...
+            ..Default::default()
+        };
+        let child = SealedHeader {
+            header: Header { gas_limit: parent.header.gas_limit, ..Default::default() },
+            ..Default::default()
+        };
+        let mut chain_spec = ChainSpec::default();
+        chain_spec.hardforks.insert(Hardfork::London, ForkCondition::Block(1));
+
+        assert_eq!(check_gas_limit(&parent, &child, &chain_spec), Ok(()));
```
Comment (Rjected) [inline crates/consensus/common/src/validation.rs:880]: this probably won't test anything related to london since parent and child both have block number zero in this case, I would set the block number of the parent, and have London fork at block zero when testing block number edge cases

**Review by Rjected — CHANGES_REQUESTED** [review]
good start! we should be able to cover this method comprehensively with a few changes

File: crates/consensus/common/src/validation.rs (hunk 1 above)
Comment (DoTheBestToGetTheBest) [reply crates/consensus/common/src/validation.rs:889]: I just added it for better readability and we can us it if we want to use SealedHeader with default header it become more easy

**Comment by DoTheBestToGetTheBest** [comment]
> good start! we should be able to cover this method comprehensively with a few changes

Ty for your help ! Appreciate lot!

File: crates/consensus/common/src/validation.rs (hunk 1 above)
Comment (Rjected) [reply crates/consensus/common/src/validation.rs:898]: we're still using `create_header` here, would recommend removing the method and creating the headers manually in tests

File: crates/consensus/common/src/validation.rs [hunk 3]
```diff
@@ -848,4 +848,87 @@ mod tests {
...
+            header: Header { gas_limit: parent.header.gas_limit + 5, ..Default::default() },
+            ..Default::default()
+        };
+        let chain_spec = ChainSpec::default();
+
+        assert_eq!(check_gas_limit(&parent, &child, &chain_spec), Ok(()));
+    }
+
+    #[test]
+    #[ignore]
```
Comment (Rjected) [inline crates/consensus/common/src/validation.rs:868]: why is this ignored?

**Review by Rjected — CHANGES_REQUESTED** [review]
have some comments remaining from the previous review, also other small things

**Review by Rjected — APPROVED** [review]
this looks good to me, ty!
